src/component/category-item/index.js

```diff
@@ -0,0 +1,39 @@
+import React from 'react';
+import CategoryForm from '../category-form';
+
+class CategoryItem extends React.Component {
+  constructor(props){
+  super(props)
+  console.log('hello there!!!', this.props.categoryDelete );
+}
+  render() {
+    return (
+      <section className='category-item'>
+        <ul>
+          {this.props.categories.map((item, i) =>
+            <li key={i}>
+              <button onClick={() => this.props.categoryDelete(item)}>DELETE</button>
+
+              <div>
+                <p>name: {item.name}</p>
+                <p>budget: {item.budget}</p>
+              </div>
+
+              <CategoryForm
+                category={item}
+                submitTitle='update category'
+                buttonText='update category'
+                handleSubmit={(category) => {
```

Comment:
handleSubmit you don't want it here, You may want to pass categoryUpdate from parent as value of onComplete props. like 
                onComplete={this.props.categoryUpdate}